### PR TITLE
[formidable] add missing options, extends class from EventEmitter

### PR DESCRIPTION
**Pull request by martin-badin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test formidable`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/node-formidable/formidable#options>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/formidable/Formidable.d.ts [hunk 1]
```diff
@@ -12,17 +18,17 @@ declare class IncomingForm {
      */
     parse(request: IncomingMessage, callback: (err: any, fields: Fields, files: Files) => void): void;
 
-    once(name: "end", callback: () => void): void;
-    once(name: "error", callback: (err: any) => void): void;
+    once(eventName: 'end', listener: () => void): this;
+    once(eventName: 'error', listener: (err: any) => void): this;
```
Comment (martin-badin) [inline types/formidable/Formidable.d.ts:22]: these methods return `this` based on the extended class

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/events.d.ts#L359

File: types/formidable/Formidable.d.ts [hunk 2]
```diff
@@ -12,17 +18,17 @@ declare class IncomingForm {
...
-    on(name: "field", callback: (name: string, value: string) => void): void;
-    on(name: "fileBegin" | "file", callback: (formName: string, file: File) => void): void;
-    on(name: "progress", callback: (bytesReceived: number, bytesExpected: number) => void): void;
-    on(name: string, callback: () => void): void;
+    on(eventName: 'data', listener: (data: EventData) => void): this;
+    on(eventName: 'error', listener: (err: any) => void): this;
+    on(eventName: 'field', listener: (name: string, value: string) => void): this;
+    on(eventName: 'fileBegin' | 'file', listener: (formName: string, file: File) => void): this;
+    on(eventName: 'progress', listener: (bytesReceived: number, bytesExpected: number) => void): this;
+    on(eventName: string, listener: () => void): this;
```
Comment (martin-badin) [inline types/formidable/Formidable.d.ts:29]: these methods return `this` based on the extended class

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/events.d.ts#L330

File: types/formidable/Formidable.d.ts [hunk 3]
```diff
@@ -12,17 +18,17 @@ declare class IncomingForm {
...
 
-    emit(name: "data", data: EmitData): void;
+    emit(eventName: 'data', data: EmitData): boolean;
```
Comment (martin-badin) [inline types/formidable/Formidable.d.ts:31]: the method returns `boolean` based on the extended class

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/events.d.ts#L555

**Comment by robahtou** [comment]
@martin-badin
Looking at the type `fileWriteStreamHandler?: (() => Writable) | undefined;`

In my formidable config I have
```
...
fileWriteStreamHandler: streamUploadImage,
...
```
where 
```
function streamUploadImage({ mimetype }): Stream {
  const passThrough = new PassThrough();
  ...
  ...
  return passThrough;
}
```
I get the TS error `Type '({ mimetype }: { mimetype: any; }) => internal' is not assignable to type '() => Writable'.`

Does the type signature for `fileWriteStreamHandler` need to be updated or am I using the stream functionality incorrectly?

**Comment by martin-badin** [comment]
> @martin-badin Looking at the type `fileWriteStreamHandler?: (() => Writable) | undefined;`
> 
> In my formidable config I have
> 
> ```
> ...
> fileWriteStreamHandler: streamUploadImage,
> ...
> ```
> 
> where
> 
> ```
> function streamUploadImage({ mimetype }): Stream {
>   const passThrough = new PassThrough();
>   ...
>   ...
>   return passThrough;
> }
> ```
> 
> I get the TS error `Type '({ mimetype }: { mimetype: any; }) => internal' is not assignable to type '() => Writable'.`
> 
> Does the type signature for `fileWriteStreamHandler` need to be updated or am I using the stream functionality incorrectly?

The function doesn't accept the parameter. I can update the function If you think that the function should accept the parameter.

**Comment by robahtou** [comment]
@martin-badin I think the function should accept the parameters
I should mention that Im using formidable@v3.2.1

**Review by gabritto — APPROVED** [review]

**Pull request merged**